src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalFilterLongReadAlignmentsSAMSpark.java

```diff
@@ -0,0 +1,328 @@
+
+        // group 1: get max aligner score of mappings to canonical chromosomes and speed up in case of too many mappings
+        final int maxCanonicalChrAlignerScore = alignedContig.alignmentIntervals.stream()
+                .filter(alignmentInterval -> canonicalChromosomes.contains(alignmentInterval.referenceSpan.getContig()))
+                .mapToInt(ai -> ai.alnScore).max().orElse(0); // possible that no mapping to canonical chromosomes
+
+        // speed up if number of alignments is too high (>10)
+        // if mapped to canonical chromosomes, MQ must be >10; otherwise, must have AS higher than max canonical aligner score
+        final List<AlignmentInterval> alignmentIntervals;
+        if (alignedContig.alignmentIntervals.size() > 10) {
+            alignmentIntervals = alignedContig.alignmentIntervals.stream()
+                    .filter(alignmentInterval -> (!canonicalChromosomes.contains(alignmentInterval.referenceSpan.getContig())
+                                                                           && alignmentInterval.alnScore > maxCanonicalChrAlignerScore)
+                                                 || alignmentInterval.mapQual>10)
+                    .collect(Collectors.toList());
+        } else {
+            alignmentIntervals = alignedContig.alignmentIntervals;
+        }
+
+        final int newMaxCanonicalChrAlignerScore = alignmentIntervals.stream()
+                .filter(alignmentInterval -> canonicalChromosomes.contains(alignmentInterval.referenceSpan.getContig()))
+                .mapToInt(ai -> ai.alnScore).max().orElse(0); // possible that no mapping to canonical chromosomes
+
+        // group 2: generate, and score configurations
+        final List<List<AlignmentInterval>> allConfigurations = Sets.powerSet(new HashSet<>(alignmentIntervals))
+                .stream().map(ArrayList::new)
+                // make sure within each configuration, alignments would be sorted as they would be in a corresponding AlignedContig
+                .map(ls -> ls.stream().sorted(AlignedContig.sortAlignments()).collect(Collectors.toList()))
+                .collect(Collectors.toList());
+
+        final List<Double> scores = allConfigurations.stream()
+                .map(configuration -> computeScoreOfConfiguration(configuration, canonicalChromosomes, newMaxCanonicalChrAlignerScore))
+                .collect(SVUtils.arrayListCollector(allConfigurations.size()));
+
+        // group 3: pick the best-scored configuration(s) (if multiple configuration equally-good scored, return all of them)
+        final double maxScore = scores.stream().mapToDouble(Double::doubleValue).max()
+                .orElseThrow(() -> new GATKException("Cannot find best-scoring configuration on alignments of contig: " + alignedContig.contigName));
+
+        final List<List<AlignmentInterval>> result = new ArrayList<>();
+        for(int i = 0; i < allConfigurations.size(); ++i) {
```

Comment:
Because I need index into two lists (hence need an index object). Replaced with a "hackish" use of IntStream.